ingest/processors/token_transfer/token_transfer_processor.go

```diff
@@ -0,0 +1,927 @@
+			diff = 50 - 0 = 50 > 0, so we issue a MINT event for 50 XLM
+
+		4.	Account only in eventsMap
+			changesMap[account, XLM] = 0	--	No changes recorded for this account (key doesn't exist)
+			eventsMap[account, XLM] = 30	--	But our events show 30 XLM being credited
+			diff = 0 - 30 = -30 < 0, so we issue a BURN event for 30 XLM
+
+			More nuanced scenarios
+
+		5.  changesMap positive, eventsMap negative
+			changesMap[account, XLM] = 100	--	Account gained 100 XLM according to ledger changes
+			eventsMap[account, XLM] = -50	--	But our events show 50 XLM being deducted
+			diff = 100 - (-50) = 150 > 0, so we issue a MINT event for 150 XLM
+
+		6.	changesMap negative, eventsMap positive
+			changesMap[account, XLM] = -80	--	Account lost 80 XLM according to ledger changes
+			eventsMap[account, XLM] = 40	--	But our events show 40 XLM being credited
+			diff = -80 - 40 = -120 < 0, so we issue a BURN event for 120 XLM
+
+			Even more nuanced scenarios
+
+		7.	Opposite of Case 2
+			changesMap[account, XLM] = -20	--	Account lost 20 XLM according to ledger changes
+			eventsMap[account, XLM] = -50	--	But our events show 50 XLM being lost
+			diff = -20 - (-50) = 30 > 0, so we issue a MINT event for 30 XLM
+
+		8. 	Opposite of Case 1
+			changesMap[account, XLM] = 70	-- 	Account gained 70 XLM according to ledger changes
+			eventsMap[account, XLM] = 100	--	But our events only account for 100 XLM
+			diff = 70 - 100 = -30, so we issue a BURN event for 30 XLM
+
+
+	*/
+
+	// Calculate the difference
+	var diff int64
+	if existsInChanges && existsInEvents {
+		// Both maps have entries for this account/asset
+		diff = changesBalance - eventsBalance
+	} else if existsInChanges {
+		// Only in changes map, not in events map
+		diff = changesBalance
+	} else if existsInEvents {
+		// Only in events map, not in changes map
+		diff = -eventsBalance
+	} else {
+		// Not in either map, no difference
+		return events
+	}
+
+	// If no difference, return empty events list
+	if diff == 0 {
+		return events
+	}
+
+	meta := p.generateEventMeta(tx, &opIndex, asset)
+	protoAsset := assetProto.NewProtoAsset(asset)
+
+	// Generate appropriate event based on the difference
+	if diff > 0 {
+		// changesMap shows more XLM than eventsMap - need to MINT
+		mintEvent := NewMintEvent(meta, accountStr, amount.String64Raw(xdr.Int64(diff)), protoAsset)
+		events = append(events, mintEvent)
+	} else {
+		// changesMap shows less XLM than eventsMap - need to BURN
+		burnEvent := NewBurnEvent(meta, accountStr, amount.String64Raw(xdr.Int64(-diff)), protoAsset)
+		events = append(events, burnEvent)
+	}
+
+	return events
+}
+
+func (p *EventsProcessor) generateXlmReconciliationEvents(tx ingest.LedgerTransaction, opIndex uint32, op xdr.Operation, operationEvents []*TokenTransferEvent) ([]*TokenTransferEvent, error) {
+	// DO not run this reconciliation check for ledgers > 8 or if operation is Inflation
+	if tx.Ledger.ProtocolVersion() > 8 {
```

Comment:
In https://github.com/stellar/stellar-protocol/blob/master/core/cap-0067.md#retroactively-emitting-events it states:
>Another relevant detail for replaying events is that prior to protocol 8, there was a bug that could result in the minting/burning of XLM. 

```suggestion
	// DO not run this reconciliation check for ledgers prior to protocol 8 or if operation is Inflation
	if tx.Ledger.ProtocolVersion() >= 8 {
```

Reply:
the if statement should also check that op.Body.Type is not xdr.OperationTypeInflation to be consistent with your comment

Reply:
nice catch. 
will make the change

Reply:
removed the check altogether from this function.
The check is not at callsite, and at the callsite (`EventsFromOperation())`, reconciliation will be skipped for `tx.Ledger.ProtocolVersion() >= 8`